self-core-impl/src/main/java/com/selfxdsd/core/projects/StripeWallet.java

```diff
@@ -96,8 +132,97 @@ public BigDecimal cash() {
     }
 
     @Override
-    public Invoice pay(final Invoice invoice) {
-        throw new UnsupportedOperationException("Not yet implemented.");
+    public Wallet pay(final Invoice invoice) {
+        if (invoice.isPaid()) {
+            throw new InvoiceException.AlreadyPaid(invoice);
+        }
+
+        final BigDecimal newLimit = this.limit.subtract(invoice.totalAmount());
+        if (newLimit.longValueExact() < 0L) {
+            throw new WalletPaymentException("No cash available in wallet "
+                + "for paying invoice #" + invoice.invoiceId()
+                + ". Please increase the limit from your dashboard with"
+                + " at least " + newLimit.abs().add(invoice.totalAmount())
+                .divide(BigDecimal.valueOf(1000), RoundingMode.HALF_UP) + "$."
+            );
+        }
+
+        ensureApiToken();
+
+        try {
+            final Contributor contributor = invoice.contract().contributor();
+            final PayoutMethod payoutMethod = this.storage
+                .payoutMethods()
+                .ofContributor(contributor)
+                .active();
+            if (payoutMethod == null) {
+                throw new WalletPaymentException(
+                    "No active payout method for contributor "
+                        + contributor.username()
+                );
+            }
+            final PaymentMethod paymentMethod = this.storage
+                .paymentMethods()
+                .ofWallet(this)
+                .active();
+            if (paymentMethod == null) {
+                throw new WalletPaymentException(
+                    "No active payment method for wallet #"
+                        + this.identifier + " of project "
+                        + this.project.repoFullName() + "/"
+                        + this.project.provider()
+                );
+            }
+            final PaymentIntent paymentIntent = PaymentIntent
+                .create(PaymentIntentCreateParams.builder()
+                    .setCurrency("usd")
+                    .setAmount(invoice.totalAmount().longValueExact())
+                    .setCustomer(payoutMethod.identifier())
+                    .setPaymentMethod(paymentMethod.identifier())
+                    .setConfirm(true)
+                    .build());
```

Comment:
@criske Which Stripe Docs did you read in order to come up with this code? :D

Reply:
@amihaiemil [Payment Intents API](https://stripe.com/docs/payments/payment-intents)
https://stripe.com/docs/api/payment_intents/create
https://stripe.com/docs/api/payment_intents/object